functions/venue.js

```diff
@@ -543,6 +544,41 @@ exports.toggleDustStorm = functions.https.onCall(async (_data, context) => {
   }
 });
 
+exports.getVenue = functions.https.onCall(async (data) => {
```

Comment:
Is this even needed anymore after the other changes? No point having a hacky 'fallback' if the standard way works fine.

Reply:
Agree on the "if the standard way works fine" I'm just not ready to call that if case true.  Some pros/cons I've noticed:


PRO: Data for event does deterministically come first (Firestore connect may decide to send it in different order)
PRO: Fallback data for venue, getting it through 2 different means and using the first available data

**CON: No significant measurable performance benefits (on localhost)**

CON: Hard to memoize AJAX call without it becoming stale, memoized can be useful only on first render.
CON If not memoized, request will go more than once.

Reply:
> CON: Hard to memoize AJAX call without it becoming stale, memoized can be useful only on first render.

Is this because we fetch it once rather than subscribing to live updates? If so, makes sense.

Reply:
Browser network tab has shown more than one request.
Sure one is right at the start, but re-render would trigger it again, and by that time the Firestore connect data should already be available (if not triggering the render)

Reply:
Makes sense

Reply:
Removed the BE function and the code in `VenuePage` that was calling it in 0da80c02d45c99456fdaff937409dc563ca3fdb4.
Reason being, too many requests, not enough performance gain.